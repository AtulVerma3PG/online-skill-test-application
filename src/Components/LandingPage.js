import React from "react";
/**
 * Handle irregular browser behaviour
 */
const LandingPage = (props) => {
  const { history } = props;
  const openHome = () => {
    history.push("/");
  };
  return (
    <div>
      <h1>Landing Page</h1>
      <button id="home" type="button" onClick={openHome} className="btn btn-primary btn-lg">
        Home
      </button>
    </div>
  );
};

LandingPage.propTypes = {
  history: Object.isRequired,
};

export default LandingPage;
