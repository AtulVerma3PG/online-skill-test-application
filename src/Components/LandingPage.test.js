/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import LandingPage from "./LandingPage";
import { createMemoryHistory } from "history";

configure({adapter: new Adapter()});
test("First Submit Test Snapshot test", () => {
  const component = renderer.create(
    <LandingPage />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const history = createMemoryHistory();
history.push('/');
describe('Test Button component', () => {
  it('Test click event', () => {
    const historyMock = { push: jest.fn() };
    const shallowCopy = shallow(<LandingPage history={historyMock}/>);
    shallowCopy.find('button').simulate('onClick');
    expect(history.location.pathname).toEqual('/')
  });
});
