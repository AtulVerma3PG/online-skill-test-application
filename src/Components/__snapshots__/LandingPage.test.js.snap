// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Submit Test Snapshot test 1`] = `
<div>
  <h1>
    Landing Page
  </h1>
  <button
    className="btn btn-primary btn-lg"
    id="home"
    onClick={[Function]}
    type="button"
  >
    Home
  </button>
</div>
`;
